pkgs/native_toolchain_c/test/cbuilder/testfiles/dynamically_linked/src/dynamically_linked.c

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#include "math.h"
+
+int main() {
+  return math_add(1, 2);
```

Comment:
It's a bit weird to expect a return code of 3 as the result of doing 1+2 because non-zero exit code usually indicate a problem. It's better to check whether the result is equal to 3 in c, and return exitcode 0 if the calculation was correct and 1 or 255 if the calculation is not correct. Or, for example printf the result and check the stdout in the test.